src/Microsoft.Framework.PackageManager/Restore/LockFileWriter.cs

```diff
@@ -0,0 +1,235 @@
+            }
+
+            if (library.CompileTimeAssemblies.Count > 0)
+            {
+                json["compile"] = WritePathArray(library.CompileTimeAssemblies, WriteString);
+            }
+
+            if (library.RuntimeAssemblies.Count > 0)
+            {
+                json["runtime"] = WritePathArray(library.RuntimeAssemblies, WriteString);
+            }
+
+            if (library.NativeLibraries.Count > 0)
+            {
+                json["native"] = WritePathArray(library.NativeLibraries, WriteString);
+            }
+
+            return new JProperty(library.Name + "/" + library.Version, json);
+        }
+
+        private JProperty WriteProjectFileDependencyGroup(ProjectFileDependencyGroup frameworkInfo)
+        {
+            return new JProperty(
+                frameworkInfo.FrameworkName,
+                WriteArray(frameworkInfo.Dependencies, WriteString));
+        }
+
+        private JToken WriteFrameworkAssemblies(IList<FrameworkAssemblyReference> frameworkAssemblies)
+        {
+            var groups = frameworkAssemblies.SelectMany(x =>
+            {
+                if (x.SupportedFrameworks.Any())
+                {
+                    return x.SupportedFrameworks.Select(xx => new { x.AssemblyName, FrameworkName = xx });
+                }
+                else
+                {
+                    return new[] { new { x.AssemblyName, FrameworkName = default(FrameworkName) } };
+                }
+            }).GroupBy(x => x.FrameworkName);
+
+            return WriteObject(groups, group =>
+            {
+                return new JProperty(group.Key?.ToString() ?? "*", group.Select(x => new JValue(x.AssemblyName)));
+            });
+        }
+
+        private void WriteFrameworkAssemblies(JToken json, string property, IList<FrameworkAssemblyReference> frameworkAssemblies)
+        {
+            if (frameworkAssemblies.Any())
+            {
+                json[property] = WriteFrameworkAssemblies(frameworkAssemblies);
+            }
+        }
+
+        private JProperty WritePackageDependencySet(PackageDependencySet item)
+        {
+            return new JProperty(
+                item.TargetFramework?.ToString() ?? "*",
+                WriteObject(item.Dependencies, WritePackageDependency));
+        }
+
+        private JProperty WritePackageDependency(PackageDependency item)
+        {
+            return new JProperty(
+                item.Id,
+                WriteString(item.VersionSpec?.ToString()));
+        }
+
+        private JToken WriteFrameworkAssemblyReference(FrameworkAssemblyReference item)
+        {
+            return new JValue(item.AssemblyName);
+        }
+
+        private JToken WritePackageReferenceSet(PackageReferenceSet item)
+        {
+            var json = new JObject();
+            json["targetFramework"] = item.TargetFramework?.ToString();
```

Comment:
Fixing `LockFileWriter` is not in the scope of this PR. This is merely copied from `LockFileFormat` to separate read and write.